app/src/main/java/com/davidtomas/taskyapp/core/domain/util/Result.kt

```diff
@@ -0,0 +1,28 @@
+package com.davidtomas.taskyapp.core.domain.util
+
+typealias RootError = Error
+
+sealed interface Result<out D, out E : RootError> {
+    data class Success<out D, out E : RootError>(val data: D) : Result<D, E>
+    data class Error<out D, out E : RootError>(val error: E?) : Result<D, E>
+
+    fun fold(
+        onError: (E?) -> Any,
+        onSuccess: (D) -> Any
+    ): Any =
+        when (this) {
+            is Error -> onError(error)
+            is Success -> onSuccess(data)
+        }
+}
+
+fun <D, E : RootError> Result<D, E>.asEmptyDataResult(): EmptyDataResult = Result.Success(Unit)
```

Comment:
So this will convert any data result, no matter if success or error, into a success result? I don't think this is what you want :D

Reply:
totally true, hahaha